Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/ControllerFinder.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Devices;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.Utilities.Solvers
+{
+    /// <summary>
+    /// ControllerFinder is a base class providing simple event handling for getting/releasing MotionController Transforms.
+    /// </summary>
+    public abstract class ControllerFinder : MonoBehaviour, IMixedRealitySourceStateHandler
```

Comment:
Isn't this the same as the ControllerPoseSynchonizer?

Reply:
That was it, I recall rejecting bring it in and conceded slightly when we changed the implementation in line with the MRTK approach.

Reply:
It's not, no. The ControllerPoseSynchonizer moves an object to a specific pose based on the events. This exposes a transform that can be used to follow or do math on or whatever.

This additionally diverges from ControllerPoseSynchonizer when specific controller attachment points come online.